vms/platformvm/vm.go

```diff
@@ -270,6 +271,31 @@ func (vm *VM) Initialize(
 	return nil
 }
 
+func (vm *VM) PruneMempool() {
+	removedTxs := set.Set[*txs.Tx]{}
+	droppedTxs := make(map[ids.ID]error)
+
+	vm.Builder.Iterate(func(tx *txs.Tx) bool {
+		_, _, dropped, err := vm.Builder.VerifyTxAgainstAcceptedState(tx)
```

Comment:
If we're going to do this periodically, I feel like we could just do it on top of the preferred state (which wouldn't require any of the builder changes afaik)